### ISSUE TITLE
[a11y] home page donate banner should have unique role/label/title

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#15000]
## Description

The donate banner on the home page is marked up as an `<aside>`, so it should have a unique role or role/label/title (i.e. accessible name) combination.

The newsletter form might also need a unique landmark identifier.

### Steps to reproduce

https://www-dev.allizom.org/en-US/

### Expected result

Each landmark should have a unique role/label/title so that screen readers can more easily distinguish each `<aside>`.

### Actual result

![image](https://github.com/user-attachments/assets/a8933afb-c95a-4d9d-8ed7-38fa85588da3)

### Environment

N/A
